lib/resources/r4/document_reference.yaml

```diff
@@ -2,12 +2,77 @@
 name: DocumentReference
 field_name_base_url: https://www.hl7.org/fhir/documentreference-definitions.html#DocumentReference
 fields:
+- name: resourceType
+  required: 'Yes'
+  type: string
+  description: The type of the FHIR resource.
+  example: |
+    {
+      "resourceType": "DocumentReference"
+    }
+  note: resourceType must be DocumentReference.
+  url: http://hl7.org/fhir/r4/documentreference.html
+
+- name: status
+  required: 'Yes'
+  type: code
+  action:
+    - create
+  description: current
+  note: The status of this document reference. Must be a supported Document Reference Status
+  url: http://hl7.org/fhir/r4/documentreference-definitions.html#DocumentReference.status
+  example: |
+    {
+      "status": "current"
+    }
+  binding:
+    description: The status of the document reference.
+    terminology:
+      - display: DocumentReference status
+        system: http://hl7.org/fhir/document-reference-status
+        info_link: https://www.hl7.org/fhir/valueset-document-reference-status.html
+
+- name: docStatus
+  required: 'Yes'
+  type: code
+  action:
+    - create
+  description: final | amended
+  note: The status of the underlying document. Must be a supported Composition Status
+  url: http://hl7.org/fhir/r4/documentreference-definitions.html#DocumentReference.docStatus
+  example: |
+    {
+      "docStatus": "final"
+    }
+  binding:
+    description: Status of the underlying document.
+    terminology:
+      - display: DocumentReference docStatus
+        system: http://hl7.org/fhir/r4/codesystem-composition-status.html
+        info_link: http://hl7.org/fhir/r4/valueset-composition-status.html
+
 - name: type
-  required: 'No'
+  required: 'Yes'
   type: CodeableConcept
-  description: Precise type of clinical document
+  action:
+    - create
+  description: Specifies the particular kind of document referenced.
+  note: Coding must include system and code, must not contain proprietary and standard codes together. If proprietary, must be from code set 72.
```

Comment:
We'll need to update this to be similar to dstu2.

Reply:
You can find the changes here: https://github.com/cerner/fhir.cerner.com/pull/446/commits/c11917e277b1bde8d54f91f63e4134060da17f1e